### PR TITLE
Added Classwise IoU Metric

### PR DESCRIPTION
**Pull request by adisingh50**

### CODE REVIEW

File: spectr/src/overfitting/overfit_train.py [hunk 1]
```diff
@@ -31,8 +32,7 @@ def main():
     y = y.cuda()
 
     # Define Loss and Optimizer
-    class_weights = get_class_weights(y)
-    ce_loss = nn.CrossEntropyLoss(weight=class_weights).cuda()
```
Comment (adisingh50) [inline spectr/src/overfitting/overfit_train.py:35]: weighted loss isn't helping us. so going back to unweighted

File: spectr/src/overfitting/overfit_train.py [hunk 2]
```diff
@@ -53,6 +53,12 @@ def main():
     # Save this model to file
     torch.save(model.state_dict(), "./overfit_model.pth")
 
+    # Create necessary directories to store visualizations, if they don't already exist
+    os.makedirs("overfit_results", exist_ok=True)
+    os.makedirs("overfit_results/images", exist_ok=True)
+    os.makedirs("overfit_results/ground_truth", exist_ok=True)
+    os.makedirs("overfit_results/pred_labels", exist_ok=True)
```
Comment (adisingh50) [inline spectr/src/overfitting/overfit_train.py:60]: auto generates folders so it can run on your machine
